components/bsx/Create/CreateToken.vue

```diff
@@ -0,0 +1,289 @@
+  protected collections: MintedCollection[] = []
+  protected postfix = true
+  protected deposit = '0'
+  protected depositPerByte = BigInt(0)
+  protected attributes: Attribute[] = []
+  protected nsfw = false
+  protected price: string | number = 0
+
+  protected updatePrice(value: number) {
+    this.price = value
+  }
+
+  get hasPrice() {
+    return Number(this.price)
+  }
+
+  public async created() {
+    onApiConnect(() => {
+      const instanceDeposit = getInstanceDeposit()
+      const metadataDeposit = getMetadataDeposit()
+      this.deposit = (instanceDeposit + metadataDeposit).toString()
+    })
+  }
+
+  @Watch('accountId', { immediate: true })
+  hasAccount(value: string, oldVal: string) {
+    if (shouldUpdate(value, oldVal)) {
+      this.fetchCollections()
+    }
+  }
+
+  public async fetchCollections() {
+    const query = await resolveQueryPath(this.urlPrefix, 'collectionForMint')
+    const collections = await this.$apollo.query({
+      query: query.default,
+      client: this.urlPrefix,
+      variables: {
+        account: this.accountId,
+      },
+      fetchPolicy: 'network-only',
+    })
+
+    const {
+      data: { collectionEntities },
+    } = collections
+
+    this.collections = unwrapSafe(collectionEntities)?.map((ce: any) => ({
+      ...ce,
+      alreadyMinted: ce.nfts?.length,
+      lastIndexUsed: Number(ce.nfts[0]?.index || 0),
+    }))
+  }
+
+  get disabled() {
+    return !(this.base.name && this.base.file && this.base.selectedCollection)
+  }
+
+  get hasSupport(): boolean {
+    return this.$store.state.preferences.hasSupport
+  }
+
+  get hasCarbonOffset(): boolean {
+    return this.$store.state.preferences.hasCarbonOffset
+  }
+
+  get arweaveUpload(): boolean {
+    return this.$store.state.preferences.arweaveUpload
+  }
+
+  protected async submit(): Promise<void> {
+    if (!this.base.selectedCollection) {
+      throw ReferenceError('[MINT] Unable to mint without collection')
+    }
+
+    this.isLoading = true
+    this.status = 'loader.ipfs'
+    const { api } = Connector.getInstance()
+    const { selectedCollection } = this.base
+    const {
+      alreadyMinted,
+      id: collectionId,
+      lastIndexUsed,
+    } = selectedCollection
+
+    try {
+      const metadata = await this.constructMeta()
+      const cb = api.tx.utility.batchAll
+      const nextId = Math.max(lastIndexUsed + 1, alreadyMinted)
+      const create = api.tx.nft.mint(collectionId, nextId, metadata)
+      const list = this.price
+        ? [api.tx.marketplace.setPrice(collectionId, nextId, this.price)]
+        : []
+
+      // const support = await canSupport(this.hasSupport)
+      //
+      const args = [[create, ...list]]
+
+      await this.howAboutToExecute(this.accountId, cb, args, (blockNumber) => {
+        showNotification(
+          `[NFT] Saved ${this.base.name} in block ${blockNumber}`,
+          notificationTypes.success
+        )
+
+        this.navigateToDetail(collectionId, String(nextId))
```

Comment:
if nothing went wrong, `this.isLoading = false`

Reply:
resolved by finally clause in d09ea107aca23bab714dc8d0a45e06f6e456271d